src/test/run-pass/diverging-fn-tail-35849.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
```

Comment:
Nice test file naming scheme! cc @rust-lang/compiler Can we decide on something like this when the testcase comes from an issue but it has some kind of categorization otherwise?